nbconvert/exporters/templateexporter.py

```diff
@@ -560,18 +560,37 @@ def get_template_names(self):
                 template_dir = os.path.join(base_dir, base_template)
                 if os.path.exists(template_dir):
                     found_at_least_one = True
-                conf_file = os.path.join(template_dir, 'conf.json')
-                if os.path.exists(conf_file):
-                    with open(conf_file) as f:
-                        conf = recursive_update(json.load(f), conf)
-            for root_dir in root_dirs:
-                template_dir = os.path.join(root_dir, 'nbconvert', 'templates', base_template)
-                if os.path.exists(template_dir):
-                    found_at_least_one = True
-                conf_file = os.path.join(template_dir, 'conf.json')
-                if os.path.exists(conf_file):
-                    with open(conf_file) as f:
-                        conf = recursive_update(json.load(f), conf)
+                    conf_file = os.path.join(template_dir, 'conf.json')
+                    if os.path.exists(conf_file):
+                        with open(conf_file) as f:
+                            conf = recursive_update(json.load(f), conf)
+                    break
```

Comment:
A minor optimization, this does change the algorithm to find the first instead of the last match. I _think_ this is fine but I'd like other eyes on it in case I need to reverse the iteration ordering.

Reply:
I am not 100% sure about the `break` statement.
What if configuration is overriden in a higher precedence extra template dir?

Ping @maartenbreddels. Let me know what you think.

Reply:
But it will not merge the configuration with other matches?

Reply:
It wasn't before because it overwrote the `conf` and `conf_file` whenever it found a template until the last one found. I don't think it changes the behavior beyond preferring the first exact match over the last exact match. If you're really worried I can remove the optimization.